src/Maestro/Maestro.Web/Helpers.cs

```diff
@@ -99,5 +102,87 @@ public static async Task<IActionResult> ProxyRequestAsync(this HttpContext conte
             }
 
         }
+
+        public static Dictionary<string, KustoQuery> CreateBuildTimesQueries(string repository, string branch, int days)
+        {
+            var parameters = new List<KustoParameter> {
+                new KustoParameter("_Repository", KustoDataTypes.String,  repository.Split("/").Last()),
+                new KustoParameter("_SourceBranch", KustoDataTypes.String, branch),
+                new KustoParameter("_Days", KustoDataTypes.TimeSpan, $"{days}d")
+            };
+
+            string prProject = "public";
+
+            // Builds in AzDo are only found in the internal project
+            if (repository.Contains("dev.azure.com"))
```

Comment:
"contains" is dangerous.  `Uri.Parse().Host ==` will prevent this from making weird answers.

Reply:
Uri.Parse() is deprecated, but I can convert the string to a uri and check its host instead of contains.